src/main/java/io/supertokens/inmemorydb/queries/TOTPQueries.java

```diff
@@ -0,0 +1,312 @@
+package io.supertokens.inmemorydb.queries;
+
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.supertokens.inmemorydb.Start;
+import io.supertokens.inmemorydb.config.Config;
+import io.supertokens.pluginInterface.RowMapper;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.exceptions.StorageTransactionLogicException;
+import io.supertokens.pluginInterface.totp.TOTPDevice;
+import io.supertokens.pluginInterface.totp.TOTPUsedCode;
+
+import static io.supertokens.inmemorydb.QueryExecutorTemplate.execute;
+import static io.supertokens.inmemorydb.QueryExecutorTemplate.update;
+
+public class TOTPQueries {
+    public static String getQueryToCreateUsersTable(Start start) {
+        return "CREATE TABLE IF NOT EXISTS " + Config.getConfig(start).getTotpUsersTable() + " ("
+                + "user_id VARCHAR(128) NOT NULL,"
+                + "PRIMARY KEY (user_id))";
+    }
+
+    public static String getQueryToCreateUserDevicesTable(Start start) {
+        return "CREATE TABLE IF NOT EXISTS " + Config.getConfig(start).getTotpUserDevicesTable() + " ("
+                + "user_id VARCHAR(128) NOT NULL," + "device_name VARCHAR(256) NOT NULL,"
+                + "secret_key VARCHAR(256) NOT NULL,"
+                + "period INTEGER NOT NULL," + "skew INTEGER NOT NULL," + "verified BOOLEAN NOT NULL,"
+                + "PRIMARY KEY (user_id, device_name)"
+                + "FOREIGN KEY (user_id) REFERENCES " + Config.getConfig(start).getTotpUsersTable()
+                + "(user_id) ON DELETE CASCADE);";
+    }
+
+    public static String getQueryToCreateUsedCodesTable(Start start) {
+        return "CREATE TABLE IF NOT EXISTS " + Config.getConfig(start).getTotpUsedCodesTable() + " ("
+                + "user_id VARCHAR(128) NOT NULL, "
+                + "code CHAR(6) NOT NULL," + "is_valid BOOLEAN NOT NULL,"
+                + "created_time_ms BIGINT UNSIGNED NOT NULL,"
+                + "expiry_time_ms BIGINT UNSIGNED NOT NULL,"
+                + "FOREIGN KEY (user_id) REFERENCES " + Config.getConfig(start).getTotpUsersTable()
+                + "(user_id) ON DELETE CASCADE);";
+    }
+
+    public static String getQueryToCreateUsedCodesExpiryTimeIndex(Start start) {
+        return "CREATE INDEX IF NOT EXISTS totp_used_codes_expiry_time_ms_index ON "
+                + Config.getConfig(start).getTotpUsedCodesTable() + " (expiry_time_ms)";
+    }
+
+    public static String getQueryToCreateUsedCodesCreatedTimeIndex(Start start) {
+        return "CREATE INDEX IF NOT EXISTS totp_used_codes_created_time_ms_index ON "
+                + Config.getConfig(start).getTotpUsedCodesTable() + " (created_time_ms DESC)";
```

Comment:
i don't think you need this.. the query below is doing it based on user_id and there will not be that many codes per user ID anyway. As it is, this index is wrong, cause we want an index on user_id, created_time_ms and not just created_time_ms.. So please remove this